### PR TITLE
added Wishlist page

### PR DESCRIPTION
**Pull request by rahulcs754**
Todo Summary:

- Added wishlist page
- added remove and added function in reducer
- create context and reducer for wishlist
- check login in cart and wishlist page then use check 
- call api call for add wishlist and remove from wishlist
- added toast in wishlist

### CODE REVIEW

File: src/ApiService/Cart/addCart.js [hunk 1]
```diff
@@ -5,42 +5,78 @@ export const addCart = async (
...
+        });
+        toast.success("Add item to cart", {
+          position: "bottom-right",
+          autoClose: 2000,
+        });
+      }
+    } catch (error) {
+      toast.warning("Oops something went wrong", {
+        position: "bottom-right",
+        autoClose: 2000,
```
Comment (kjais1720) [inline src/ApiService/Cart/addCart.js]: you can configure the `<Toastcontainer/>` with these properties if you are using the same property for the toasts everywhere

File: src/ApiService/Cart/addCart.js [hunk 2]
```diff
@@ -5,42 +5,78 @@ export const addCart = async (
...
+        toast.success("Increase item quantity in the cart", {
+          position: "bottom-right",
+          autoClose: 2000,
+        });
+      }
+    } catch (error) {
+      toast.warning("Oops Something went wrong", {
         position: "bottom-right",
         autoClose: 2000,
       });
```
Comment (kjais1720) [inline src/ApiService/Cart/addCart.js]: same as above

File: src/ApiService/Wishlist/addWishlist.js [hunk 3]
```diff
@@ -1,21 +1,73 @@
...
+        ProductDispatch({ type: "IS_LIKED", payload: productDetails._id });
+        toast.success("Add item to wishlist", {
+          position: "bottom-right",
+          autoClose: 2000,
+        });
+      }
+    } catch (error) {
+      toast.warning("Oops something went wrong", {
+        position: "bottom-right",
+        autoClose: 2000,
```
Comment (kjais1720) [inline src/ApiService/Wishlist/addWishlist.js]: same as above

File: src/App.js [hunk 4]
```diff
@@ -2,13 +2,16 @@ import "react-toastify/dist/ReactToastify.css";
 import Routespaths from "./routes";
 import Header from "./Components/Header";
 import { ToastContainer } from "react-toastify";
+import { useTheme } from "./Context/ThemeContext";
+
 const App = () => {
+  const { theme, themeHanlder } = useTheme();
```
Comment (kjais1720) [inline src/App.js:8]: there's a typo

File: src/ApiService/Cart/addCart.js [hunk 5]
```diff
@@ -5,42 +5,78 @@ export const addCart = async (
...
-    if (status === 200 || status === 201) {
-      CartDispatch({ type: "ADD_TO_CART", payload: cart });
+  const findItem = CartState.cart.find((item) => item._id === product._id);
 
-      ProductDispatch({
-        type: "DECREASE_PRODUCT_QTY",
-        payload: product._id,
-      });
+  if (findItem === undefined) {
+    try {
```
Comment (therajatg) [inline src/ApiService/Cart/addCart.js:14]: undefined is false. so u can do if(!findItem) only.

File: src/ApiService/Cart/addCart.js [hunk 6]
```diff
@@ -5,42 +5,78 @@ export const addCart = async (
...
+      const {
+        data: { cart },
+        status,
+      } = await axios.post(
```
Comment (therajatg) [inline src/ApiService/Cart/addCart.js:18]: I like this destructuring part. I was using response.data.cart.

File: src/Provider/index.js [hunk 7]
```diff
@@ -3,18 +3,21 @@ import { AuthProvider } from "../Context/AuthContext";
...
+          <FilterProvider>
+            <WishProvider>
+              <CartProvider>{children}</CartProvider>
+            </WishProvider>
+          </FilterProvider>
+        </ProductProvider>
+      </AuthProvider>
+    </ThemeProvider>
   );
 };
```
Comment (therajatg) [inline src/Provider/index.js:23]: I liked that you did abstraction in index.js itself

**Review by therajatg — APPROVED** [review]
everything else looks good

File: src/App.js (hunk 4 above)
Comment (rahulcs754) [reply src/App.js:8]: thanks point out this. you will see this next commit

File: src/ApiService/Wishlist/addWishlist.js (hunk 3 above)
Comment (rahulcs754) [reply src/ApiService/Wishlist/addWishlist.js]: Thanks to give this information. i will change next commit

File: src/ApiService/Cart/addCart.js (hunk 2 above)
Comment (rahulcs754) [reply src/ApiService/Cart/addCart.js]: done as you said. Thanks for pointing

File: src/ApiService/Cart/addCart.js (hunk 1 above)
Comment (rahulcs754) [reply src/ApiService/Cart/addCart.js]: Thanks for pointing out this i will change next commit

File: src/ApiService/Cart/addCart.js (hunk 6 above)
Comment (rahulcs754) [reply src/ApiService/Cart/addCart.js:18]: thanks

File: src/ApiService/Cart/addCart.js (hunk 5 above)
Comment (rahulcs754) [reply src/ApiService/Cart/addCart.js:14]: thanks for this information

**Review by kjais1720 — APPROVED** [review]
Looks good now 👍

**Pull request merged**